### ISSUE TITLE
Recommendations example

### ISSUE THREAD

**Issue by saeta** [tensorflow/swift-models#151]
It'd be great to have an example for neural recommendations using embeddings incorporated into our model repository.

**Comment by rsdel2007** [comment]
I would like to contribute for recommendation example.

**Comment by saeta** [comment]
Hi @rsdel2007 ! Thanks for taking this on. I recommend as a starting point the neural collaborative filtering model from MLPerf. There are reference models available at: https://github.com/tensorflow/models/blob/master/official/recommendation/README.md and https://github.com/mlperf/training/blob/master/recommendation/pytorch/ncf.py

In order to ensure you have a correct model, please be sure to ensure it converges to the expected accuracy in the expected number of epochs. Thanks!

**Comment by nikochiko** [comment]
@rsdel2007 , @saeta , is this issue still open? I would like to contribute.

**Comment by ayush29** [comment]
Interested.

**Comment by SumanSudhir** [comment]
@BradLarson @saeta  can you please review my code for preparing MovieLens dataset [Here](https://github.com/SumanSudhir/NeuMF-Swift/blob/data/data/MovieLens.swift) . There is also a notebook in the same folder to run and visualize the training data

**Comment by BradLarson** [comment]
@SumanSudhir - That looks like a great start. You might examine the possibility of incorporating that with [Batcher](https://github.com/tensorflow/swift-models/tree/master/Batcher) as we migrate towards that in the datasets here.

Stylistically, there are a number of small changes I'd suggest, such as relying more heavily on type inference and not explicitly declaring types (such as in the let statement [here](https://github.com/SumanSudhir/NeuMF-Swift/blob/data/data/MovieLens.swift#L50)), or using more descriptive and more Swift-like names than `user_pool`, `df`, etc.

When you have that matched up with a model that trains using it, we'd be glad to review that in more detail within a pull request to swift-models.

**Comment by SumanSudhir** [comment]
@BradLarson I am able to train NeuMF model. You can run it by cloning the repo and running the command `swift build` and `swift run`
Since it is taking considerable time to run even one epoch on CPU so I need to train it on GPU. Can you please have a look at the [code](https://github.com/SumanSudhir/NeuMF-Swift.git) and test it

**Comment by BradLarson** [comment]
PR #406 addresses this, so unless there are further permutations we wanted to explore, we can probably close this for now.
